File: src/main/java/com/bma/algorithms/adventofcode2020/python/day3.py
f = open ("../inputs/day3.txt")

frame = []
for line  in f:
    col = []
    for char in line:
        col.append(char)
    
    frame.append(col)

slopes = [[3,1]]

count = 0
for slope in slopes:
    x = slope[0]
    y = slope[1]

    while x < len(frame):
        if frame[x][y % len(frame[x])]:
            count += 1

        y += slope[1]
        x += slope[0]

    print(count)

File: src/main/java/com/bma/algorithms/adventofcode2020/day3.cpp
#include<iostream>
#include<fstream>
#include<string>
#include<vector>

using namespace std;

int  main() {
    ifstream day_3_input("inputs/day3.txt");
    string line;

    vector<vector<char>> frame;
    while (getline(day_3_input, line)) {
        vector<char> col;
        for (int j=0; j < line.length(); j++) {
            col.push_back(line[j]);
        }
        frame.push_back(col);
    }

    // vector<vector<int>> slopes =  { {3,1} };
    vector<vector<int>> slopes =  { {1,1}, {3,1}, {5,1}, {7,1}, {1,2} };

    int mul = 1;
    
    for (auto slope: slopes) {
        int count = 0;
        int x = slope[1], y = slope[0];
        while (x < frame.size()) {
            if (frame[x][y % frame[x].size()] == '#') count++;
            y += slope[0];
            x += slope[1];
        }

        cout << count << "," << endl;
        mul *= count;
    }
    

    cout << mul;

    day_3_input.clear();
    day_3_input.close();

    return 0;
}